pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -79,6 +87,33 @@ func NewDockerRegistryServiceController(cl kclientset.Interface, options DockerR
 	e.servicesSynced = e.serviceController.HasSynced
 	e.syncRegistryLocationHandler = e.syncRegistryLocationChange
 
+	e.routeCache, e.routeController = framework.NewInformer(
```

Comment:
This is cray cray.  External registry names are special, and we need a better mechanism to do so than looking for special routes.